### PR TITLE
Add test demonstrating how trivia is handled

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs [hunk 1]
```diff
@@ -2540,8 +2519,63 @@ void M(int i, int j)
...
+
+                class C
+                {
+                    void M(int i, int j)
+                    {
+                        int[][] r =
+                        [
+                            // Leading
+                            new[] { 1 }, // Trailing
+                            new int[] { 2 }
```
Comment (sharwell) [inline src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs:2557]:
```suggestion
                            new int[] { 2 },
```

File: src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs [hunk 2]
```diff
@@ -2540,8 +2519,63 @@ void M(int i, int j)
...
+                
+                class C
+                {
+                    void M(int i, int j)
+                    {
+                        int[][] r =
+                        [
+                            // Leading
+                            [1], // Trailing
+                            [2]
```
Comment (sharwell) [inline src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs:2573]:
```suggestion
                            [2],
```

**Review by sharwell — COMMENTED** [review]
It looks like the fixed code is missing a trailing comma in two cases.

**Comment by CyrusNajmabadi** [comment]
currently our features do not add a trailing comma, unless they're adding to an existing initializer that already had one. 

I'm *open* to updating this decision and adding trailing commas in more places.  *However*, i would like to do that *outside of* this PR.

@sharwell  if we did go this route, would you be ok with these rules:

1. if we are adding to an existing collection-initializer/expression, we will have a trailing comma or not, depending on what the original code had.
2. If we are creating a new collection initializer/expression, and it all will be single-line, then we do not add a trailing comma.
3. If we are creating a new collection initializer/expression and there would be multiple elements, we place a comma after the last one.

If that fits what you want, i can make a followup PR to encode those rules.

Thanks!

**Comment by sharwell** [comment]
Yes, those are exactly the expected rules. I believe we settled on them in an earlier discussion.
https://github.com/dotnet/roslyn/pull/54859#issuecomment-1015853288

**Comment by CyrusNajmabadi** [comment]
> We should prioritize that work, as the current refactoring produces code which is problematic for maintainability.

Sure.  I'll do tomorrow.

**Review by akhera99 — APPROVED** [review]

**Pull request merged**